api.js

```diff
@@ -1119,6 +1230,13 @@
           };
         }
 
+        if(hash&&cacheRoute) {
```

Comment:
If cache was enabled earlier on, then these variables will be defined & hold values. It's not a quirk, it's plain old undefined or set.